### ISSUE TITLE
vendor vault UI's bindata_assetfs.go like Consul does?

### ISSUE THREAD

**Issue by arianvp** [hashicorp/vault#5622]
**Is your feature request related to a problem? Please describe.**
https://nixos.org/ packages  hashicorp tooling by building them from source (We are source based linux distro, like Gentoo) .   For consul, we can just use
our standard `go` build infrastructure to build  consul (we simply type `go build` in a container). And we get consul + consul UI because the consul UI assets are vendored with `https://github.com/hashicorp/consul/blob/master/agent/bindata_assetfs.go` .  This makes packaging `consul` very easy for us (https://github.com/NixOS/nixpkgs/pull/48714).

Recently, we've tried to update the `vault` package to a new version, but ran into some problems with building it. for the vault repo  this `bindata_assetfs.go` is _not_ commited.   thus a simple `go build` does not succesfully build `vault`.  

We have to build this file manually, using the yarn project in `/ui`. However, yarn is not so well-supported in NixOS and thus it has been a bit of a pain to get `vault` to build for me (https://github.com/NixOS/nixpkgs/pull/49082).   

**Describe the solution you'd like**
Just commit the resulting `bindata_assetfs.go`. Then people can simply build `vault` with a `go build`. This would make building vault a lot easier

**Describe alternatives you've considered**
I've tried building the yarn project myself but this adds  thousands of new dependencies to the nixos project, which we would prefer not to add.  (We have to package up each individual yarn package that the UI Depends on into the OS, to successfully build the binary,). We're a bit reluctant to do so.

**Comment by bricewge** [comment]
@chrishoffman Could it be possible then, to upload the build artifact as an assets on releases tags as suggested [here](https://github.com/NixOS/nixpkgs/pull/49082#issuecomment-433599774)?

**Comment by jefferai** [comment]
I looked through that issue and it doesn't seem like our suggestion to use our Dockerfile was ever brought up. Can nix use Docker to build?

**Comment by bricewge** [comment]
No, Nix can't use Docker to build a package since it is meant to build [reproducible](https://reproducible-builds.org/) packages and dockerfiles have side effects `apt-get update`, for [example](https://github.com/hashicorp/vault/blob/66ef59957aafa64bb8fdd7a01558b978e95c0e3d/scripts/cross/Dockerfile#L3). Otherwise we would have gone that way, as suggested.
